R/backport_linter.R

```diff
@@ -0,0 +1,83 @@
+#' @describeIn linters that checks for usage of unavailable functions. Not reliable for testing r-devel dependencies.
+#' @export
+backport_linter <- function(r_version = "4.0.3") {
+  function(source_file) {
+    if (is.null(source_file$xml_parsed_content) || r_version >= "r-devel") return(list())
+
+    xml <- source_file$xml_parsed_content
+
+    #browser()
+
+    names_xpath <- "//*[self::SYMBOL or self::SYMBOL_FUNCTION_CALL]"
+    all_names_nodes <- xml2::xml_find_all(xml, names_xpath)
+    all_names <- xml2::xml_text(all_names_nodes)
+
+    # guaranteed to include 1 by early return above; which.min fails if all TRUE (handled by nomatch)
+    needs_backport_names <- backports[1:(match(FALSE, r_version < names(backports), nomatch = length(backports)) - 1L)]
+
+    # not sapply, which may over-simplify -- cbind makes sure apply works
+    needs_backport <- do.call(cbind, lapply(needs_backport_names, function(nm) all_names %in% nm))
+    bad_idx <- apply(needs_backport, 1L, any)
```

Comment:
`rowSums(needs_backport) > 0L` is much faster

```
> microbenchmark::microbenchmark(
+   apply(matrix(c(T, F, T, F), nrow = 2), 1, any),
+   rowSums(matrix(c(T, F, T, F), nrow = 2)) > 0L
+ )
Unit: microseconds
                                           expr  min   lq   mean median    uq  max neval
 apply(matrix(c(T, F, T, F), nrow = 2), 1, any) 11.3 11.6 12.189  11.80 12.05 33.1   100
  rowSums(matrix(c(T, F, T, F), nrow = 2)) > 0L  3.3  3.5  3.971   3.75  4.10 17.7   100
```

Reply:
thanks... it's a shame that's the case, huh :)